### ISSUE TITLE
Async cache regen

### ISSUE THREAD

**Issue by only-a-ptr** [RigsOfRods/rigs-of-rods#2576]
RoR's modcache (content database) is central - it lists all installed terrains/actors/skins and lets player query&pick via SelectorUI. All spawned actors (vehicle/load...) keep a poiner to their CacheEntry for info and respawning.

Presently, modcache [re]build is a sequential process, using OGRE resource system:
1. resource group RGN_CONTENT is created with all locations (note RGN_TEMP is an overlap/duplicate of some sort) [Code](https://github.com/RigsOfRods/rigs-of-rods/blob/master/source/main/resources/ContentManager.cpp#L246)
2. resource bundles (ZIP/SKINZIP) are listed using OGRE ResourceGroupManager. [Code](https://github.com/RigsOfRods/rigs-of-rods/blob/master/source/main/resources/CacheSystem.cpp#L156)
3. for each bundle, a working RG is created, UI progress window is updated (frame is rendered), relevant files are loaded via Ogre::DataStream interface, parsed and indexed. [Code](https://github.com/RigsOfRods/rigs-of-rods/blob/master/source/main/resources/CacheSystem.cpp#L995)

@paroj I'd like to parallelize the above process using OGRE WorkQueue, but I'm not sure which OGRE_THREAD_SUPPORT I need (currently we use '3') and which parts of API are safe to use concurrently - The [manual](https://ogrecave.github.io/ogre/api/latest/background-resource.html) doesn't mention it. Based on https://github.com/OGRECave/ogre/issues/454, my best bet for the scenario above is this (using OGRE_THREAD_SUPPORT '3'):
1. main thread
2. main thread
3. create RGs on main thread and post tasks to work queue. read&parse files in parallel, register CacheEntries and destroy RGs on main thread again.

Is this design correct/optimal please?

**Comment by paroj** [comment]
yes, the design is sound. also see https://forums.ogre3d.org/viewtopic.php?t=95502

With OGRE_THREAD_SUPPORT=3 the API is not guarded for concurrency i.e. you are responsible for creating & locking mutexes as needed (e.g. concurrent reads are fine).

However, this rings a bell:

> 3. for each bundle, a working RG is created, UI progress window is updated (frame is rendered)

The window update might be a bottleneck. E.g. this change reduced loading the SampleBrowser from several seconds to instant:
https://github.com/OGRECave/ogre/commit/d72d5bc28091d1828605b4a543251bbbfbb23ff0
